add capability to use allfields sql notation (#11846)
doctrine/doctrine2: docs/en/reference/dql-doctrine-query-language.rst

If two arguments have the same name, a ``DuplicateFieldException`` is thrown.
If a field cannot be matched with a property name, a ``NoMatchingPropertyException`` is thrown. This typically happens when using functions without aliasing them.

> In a Dto, if you want add all fields of an entity, you can use AllFields notation :

Reviewer: ```suggestion
In a Dto, if you want add all fields of an entity, you can use ``.*`` :
```

.. code-block:: php

    <?php
    $query = $em->createQuery('SELECT NEW NAMED CustomerDTO(c.name, NEW NAMED AddressDTO(a.*) AS address) FROM Customer c JOIN c.address a');
    $users = $query->getResult(); // array of CustomerDTO

    // CustomerDTO => {name : 'DOE', email: null, city: null, address: {id: 18, city: 'New York', zip: '10011'}}

> It's recommended to use named arguments DTOs with the ``.*`` because argument order is not guaranteed.

Reviewer: ```suggestion
It's recommended to use named arguments DTOs with the ``.*`` notation because argument order is not guaranteed.
```

> It's recommended to use named arguments Dto with AllFields notation because argument order is not guaranteed.

Reviewer: ```suggestion
It's recommended to use named arguments DTOs with the ``.*`` notation because argument order is not guaranteed.
```
